### PR TITLE
fix: flaky ping test and log correction

### PR DESCRIPTION
**Pull request by aloknerurkar**
- Adding a new HostFactory option to libp2p to enable test host creation for unit tests.
- Ping test now works
-  Correcting log msg #2343

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2349)

### CODE REVIEW

File: pkg/p2p/libp2p/libp2p.go [hunk 1]
```diff
@@ -93,6 +93,7 @@ type Options struct {
 	LightNodeLimit int
 	WelcomeMessage string
 	Transaction    []byte
+	HostFactory    func(context.Context, ...libp2p.Option) (host.Host, error)
```
Comment (janos) [inline pkg/p2p/libp2p/libp2p.go:96]: Somehow, this field does not seems useful for the exposed package api, only for the tests.

Would you consider having an unexported field in `hostFactory func(context.Context, ...libp2p.Option) (host.Host, error)` instead to be set only by a new helper function defined in export_test.go. This is just a suggestion, not a hard requirement for the PR approval.

**Review by janos — APPROVED** [review]
Nice, I have only one suggestion, only if you think that it would improve exposed package api.

**Review by acud — APPROVED** [review]
Nice! thanks!

**Comment by janos** [comment]
Thank you @aloknerurkar for additional change. That was exactly what I meant. 🙏

**Pull request merged**